### ISSUE TITLE
How to prepare dataset for pyannote-audio.

### ISSUE THREAD

**Issue by alamnasim** [pyannote/pyannote-audio#180]
Hi,
I have Hindi-English mixed audio files with many speakers, if i want to train pyannote-audio for speaker diarization on my own dataset what should i do?
Can anyone explain me steps to prepare dataset for this?

I tried pre-trained model on my dataset, it is not giving good result because of domain change, so i need to train model on my own dataset.

I went through the code, i did not understand the RTTM and UEM file format.

Any help appreciated.
Thanks.

**Comment by hbredin** [comment]
Thanks for your interest in `pyannote.audio`.

I should probably write a proper tutorial to answer your question: it would be very useful for others too. 
Would you be interested in helping me in the process? Basically, you would follow the instructions in this new tutorial and let me know when something is not clear.

First question (sorry if that is obvious): is (at least part of) your dataset manually annotated in terms of "who speaks when" so you can use it as training set?

**Comment by alamnasim** [comment]
Thanks for the reply.
Yes, I am interested in the tutorial for pyannote.audio.

Yes, My training data is manually annotated as  ".npy" feature.

What would be the exact format of the annotated dataset?

**Comment by hbredin** [comment]
Here we go: https://github.com/pyannote/pyannote-audio/tree/tutorial_own_data/tutorials/own_data/train

Looking forward to your feedback!

**Comment by hbredin** [comment]
Any update on this tutorial ? Did it help you use your dataset for training?

**Comment by alamnasim** [comment]
> Here we go: https://github.com/pyannote/pyannote-audio/tree/tutorial_own_data/tutorials/own_data/train
> 
> Looking forward to your feedback!

Thanks for your reply and for wonderful tutorial.

**Comment by alamnasim** [comment]
I have calls having start time, end time and duration of every speaker in the call. I have not tried this, as was busy in some other work. I have a doubt supose i have a call with two speakers, then how do i denote speaker 2. 
Do i need to annotate in this order:
file1 1 0.0 sec 10 sec
file2 1 10 sec 10 sec
file1 1 20 sec 10 sec
file2 1 30 sec 10 sec
If file1 denote speaker1 and file2 denote speaker 2 then how do i distinguish two calls.
If i take each speaker utterance and annote them in the format of yours then how do a retain speaker change.

**Comment by hbredin** [comment]
The UEM file is not meant to store "who speaks when" annotation. 
The RTTM file is. Don't confuse the two -- they have two different roles...
Please read the tutorial carefully and let me know what is unclear...

I do have a few questions on your use case. 
Do you have two audio files for each call? One for each side of the conversation? 
Is there only one speaker per file? 
If so I guess you don't really need to do speaker diarization right?

**Comment by alamnasim** [comment]
> The UEM file is not meant to store "who speaks when" annotation.
> The RTTM file is. Don't confuse the two -- they have two different roles...
> Please read the tutorial carefully and let me know what is unclear...
> 
> I do have a few questions on your use case.
> Do you have two audio files for each call? One for each side of the conversation?
> Is there only one speaker per file?
> If so I guess you don't really need to do speaker diarization right?

Ok, I will look into tutorial again.
Yes, I have few audio files for each speaker and some of the audio files are having two speakers and I need diarization for these calls.

**Comment by haha010508** [comment]
I put my AMI data at sounddata folder and write this statement  at database.yml  
AMI: /mnt/****/*****/sounddata/amicorpus/*/audio/{uri}.wav

and run this code: 
$ pyannote-change-detection apply ${TRAIN_DIR}/weights/0870.pt AMI.SpeakerDiarization.MixHeadset precomputed/scd

but the result is:
raise ValueError(msg.format(uri=uri))
ValueError: Could not find file "ES2003a.Mix-Headset".

so what is problem in my code?

**Comment by hbredin** [comment]
@haha010508 please open a new issue as this problem is not related to the original one.

**Comment by aidosRepoint** [comment]
> Here we go: https://github.com/pyannote/pyannote-audio/tree/tutorial_own_data/tutorials/own_data/train
> 
> Looking forward to your feedback!

Hi!

the link to the tutorial is no longer available. Is it transferred somewhere else?

**Comment by hbredin** [comment]
Please refer to [this](https://github.com/pyannote/pyannote-audio) page to locate the tutorials.

**Comment by julianus-kath** [comment]
Hi there, im working on a speaker diarization task where the standard models aren't working that well, so similarly to @alamnasim , I would need to load my own data and can't use the AMI dataset. 

Unfortunately the tutorials dont really show how to load your own data to fine tune the model on; how can this be done?

**Comment by FrenchKrab** [comment]
You have to define your own "protocols" through a yml file.
You can take a look at the pyannote-database readme, especially the speaker diarization part:
- [pyannote.database SpeakerDiarization documentation](https://github.com/pyannote/pyannote-database#speaker-diarization)

It might be easier to see how they look in practice, some example database.yml files are available:
- [AMI dataset for pyannote](https://github.com/pyannote/AMI-diarization-setup)
- [datasets-pyannote](https://github.com/FrenchKrab/datasets-pyannote)

Basically you just have to write the pyannote-database yml file which indicates where to find audio files/RTTMs/UEMs, and what goes into train/dev/test sets.

**Comment by sidhamen** [comment]
I am not able to find the code for create diarization data from scratch can anyone please help me find the tutorial notebook?

**Comment by swnastya** [comment]
In the [AMI dataset for pyannote](https://github.com/pyannote/AMI-diarization-setup), there are also LAB files which are containing annotations for speech intervals. Shall I also create them for tuning the diarization pipeline, or are they for something else? I can see that LABLoader does exist along with the other loaders. But I cannot find information about LAB files in the tutorials, unfortunately; only RTTMs/UEMs are mentioned in the context of diarization.
Thank you in advance.

**Comment by WGS-note** [comment]
Hello @hbredin ,

First of all, thank you very much for open-sourcing this excellent project.

I have integrated and decoupled pyannote.audio 4.0.4 into my own speech processing pipeline, and I would like to ask for some advice regarding customization and fine-tuning.

My questions are:

1. What is the recommended way to fine-tune pyannote.audio models? For example, if I want to fine-tune speech segmentation, speaker change detection, or overlap detection, what training pipeline should I follow?

2. In overlapping speech scenarios, what approaches would you recommend to obtain better segmentation results? Are there specific strategies, models, or post-processing methods that can improve overlap handling?

Thank you very much for your time and help.
